src/components/SecondaryPanes/Frames.js

```diff
@@ -72,14 +100,24 @@ const Frames = React.createClass({
         className: classNames("frame", {
           "selected": selectedFrame && selectedFrame.id === frame.id
         }),
-        onMouseDown: () => selectFrame(frame),
+        onMouseDown: (e) => this.onMouseDown(e, frame, selectedFrame),
+        onContextMenu: (e) => this.onContextMenu(e, frame),
         tabIndex: 0
       },
       renderFrameTitle(frame),
       renderFrameLocation(frame)
     );
   },
 
+  onMouseDown(e, frame, selectedFrame) {
+    if (e.nativeEvent.which == 3 
+        && selectedFrame
+        && selectedFrame.id != frame.id) {
+      return;
```

Comment:
this was my solution to flowtype complaining about selectedFrame being possibly undefined (which is valid, user doesn't need to select a frame)